src/structui/ui.py

```
@@ -293,6 +295,14 @@ def handler(e):
                         elif prop_type in ['number', 'float'] and val is not None and val != '':
                             try: val = float(val)
                             except ValueError: pass
+                        elif not prop_type and val is not None and val != '':
+                            if isinstance(orig_val, int) and type(orig_val) is not bool:
+                                try:
+                                    if float(val).is_integer():
+                                        val = int(float(val))
+                                    else:
+                                        val = float(val)
+                                except ValueError: pass
```

Ruff (0.15.13)

[error] Multiple statements on one line (colon)

(E701)